### PR TITLE
[client/rest]: more rosetta integration changes 

### PR DESCRIPTION
**Pull request by Jaguar0625**
[client/rest]: filter out zero transfer operations
    
     problem: zero transfer operations do not change state
    solution: do not create zero transfer operations

    [client/rest]: calculate transaction location in OperationParser and pass down to lookupCurrency
    
     problem: transaction location is needed for proper mosaic alias resolution
    solution: calculate it in OperationParser

### CODE REVIEW

File: client/rest/src/plugins/rosetta/OperationParser.js [hunk 1]
```diff
@@ -161,15 +181,17 @@ export class OperationParser {
 				// due to async nature, operations from previous transaction might not be added yet
 				// so simply using operations.length would yield wrong result
 				// instead, renumber the operations below
-				const subOperations = await this.parseTransactionInternal(0, subTransaction.transaction);
+				const subOperations = await this.parseTransactionInternal(
+					subTransaction.transaction,
+					subTransaction.meta && undefined !== subTransaction.meta.index
```
Comment (Wayonb) [inline client/rest/src/plugins/rosetta/OperationParser.js:186]: Just wondering when you will not be present. 🤔

**Review by Wayonb — APPROVED** [review]
👍🏾

File: client/rest/src/plugins/rosetta/OperationParser.js (hunk 1 above)
Comment (Jaguar0625) [reply client/rest/src/plugins/rosetta/OperationParser.js:186]: when JSON is derived from sdk model

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**